### PR TITLE
Adding tests of libkcapi

### PR DESCRIPTION
**Pull request by nasastry**
security/libkcapi-tests.py adds libkcapi tests

Signed-off-by: Nageswara R Sastry <rnsastry@linux.vnet.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
Output of `avocado run libkcapi-tests.py -m libkcapi-tests.py.data/test_name.yaml`

```
JOB ID     : 69ef59af5e79f33d2a856d9d2ab69e2ec95c160c
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T23.56-69ef59a/job.log
 (1/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-test.sh-7d88:  PASS (215.10 s)
 (2/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-convenience.sh-4a4f:  PASS (42.70 s)
 (3/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-dgst-test.sh-71f4:  PASS (71.35 s)
 (4/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test.sh-5b2f:  PASS (67.01 s)
 (5/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test-large.sh-5e4c:  PASS (39.72 s)
 (6/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-hasher-test.sh-e0d3:  PASS (59.06 s)
 (7/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-fuzz-test.sh-4685:  PASS (550.25 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-09-24T23.56-69ef59a/results.html
JOB TIME   : 1048.07 s
```

File: security/libkcapi-tests.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        git.get_repo(url, destination_dir=self.workdir)
+        os.chdir(self.workdir)
+        process.run("autoreconf -i")
+        process.run("./configure --enable-kcapi-test --enable-kcapi-speed \
+                    --enable-kcapi-hasher --enable-kcapi-rngapp \
+                    --enable-kcapi-encapp --enable-kcapi-dgstapp")
+        build.make(self.workdir)
+        self.test_dir = os.path.join(self.workdir, 'test')
+        os.chdir(self.test_dir)
+        self.test_name = self.params.get('test_name', default='bash test.sh')
```
Comment (harish-24) [inline security/libkcapi-tests.py:49]: Considering `bash` is anyways required to run, it is better to append in script rather than to take as an input.

**Comment by nasastry** [comment]
@harish-24 added 'bash' as part of script and sent the updated code. Please review.

```
JOB ID     : 232937ffad752d443ff96258412cd80c998d0f32
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-09-29T05.39-232937f/job.log
 (1/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-test.sh-9d21:  PASS (273.97 s)
 (2/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-convenience.sh-3236:  PASS (77.82 s)
 (3/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-dgst-test.sh-6f25:  PASS (92.11 s)
 (4/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test.sh-cb9f:  PASS (91.72 s)
 (5/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test-large.sh-adb4:  PASS (78.08 s)
 (6/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-hasher-test.sh-4245:  PASS (81.91 s)
 (7/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-fuzz-test.sh-613d:  PASS (950.08 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-09-29T05.39-232937f/results.html
JOB TIME   : 1646.61 s
```

File: security/libkcapi-tests.py [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        git.get_repo(url, destination_dir=self.workdir)
+        os.chdir(self.workdir)
+        process.run("autoreconf -i")
+        process.run("./configure --enable-kcapi-test --enable-kcapi-speed \
+                    --enable-kcapi-hasher --enable-kcapi-rngapp \
+                    --enable-kcapi-encapp --enable-kcapi-dgstapp")
+        build.make(self.workdir)
+        self.test_dir = os.path.join(self.workdir, 'test')
+        os.chdir(self.test_dir)
+        self.test_name = self.params.get('test_name', default='test.sh')
```
Comment (harish-24) [inline security/libkcapi-tests.py:49]: this does not need to be a class variable. Also you could squash both into a single variable if that works out.

**Comment by nasastry** [comment]
@harish-24 removed the class variable and squashed in to a single variable. Thanks!!

```
JOB ID     : 1004dbb34cb4c71fd4f3593ef92ef0745c3335d9
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2020-09-29T06.31-1004dbb/job.log
 (1/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-test.sh-9d21:  PASS (276.13 s)
 (2/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-convenience.sh-3236:  PASS (77.90 s)
 (3/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-dgst-test.sh-6f25:  PASS (92.36 s)
 (4/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test.sh-cb9f:  PASS (91.87 s)
 (5/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test-large.sh-adb4:  PASS (77.99 s)
 (6/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-hasher-test.sh-4245:  PASS (88.39 s)
 (7/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-fuzz-test.sh-613d:  PASS (948.82 s)
RESULTS    : PASS 7 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2020-09-29T06.31-1004dbb/results.html
JOB TIME   : 1654.42 s
```

**Review by harish-24 — APPROVED** [review]

File: security/libkcapi-tests.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        Install the basic packages to support libkcapi
+        '''
+        # Check for basic utilities
+        smm = SoftwareManager()
+        deps = ['gcc', 'make', 'autoconf']
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://github.com/smuellerDD/libkcapi"
+        git.get_repo(url, destination_dir=self.workdir)
```
Comment (PraveenPenguin) [inline security/libkcapi-tests.py:40]: please use zip version of download  files , if it is available that help to use asset_fetch

**Review by PraveenPenguin — CHANGES_REQUESTED** [review]

**Comment by nasastry** [comment]
RHEL:
```
Fetching asset from libkcapi-tests.py:LibKCAPI.test
JOB ID     : edd1ec88ed3d837a5cd5ee4fc34e1d337df2c292
JOB LOG    : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T13.32-edd1ec8/job.log
 (1/7) libkcapi-tests.py:LibKCAPI.test;test.sh-9d21: FAIL: 936 test(s) failed, please refer to the log (33.97 s)
 (2/7) libkcapi-tests.py:LibKCAPI.test;kcapi-convenience.sh-3236: FAIL: 1 test(s) failed, please refer to the log (13.53 s)
 (3/7) libkcapi-tests.py:LibKCAPI.test;kcapi-dgst-test.sh-6f25: FAIL: 320 test(s) failed, please refer to the log (25.71 s)
 (4/7) libkcapi-tests.py:LibKCAPI.test;kcapi-enc-test.sh-cb9f: FAIL: 343 test(s) failed, please refer to the log (25.03 s)
 (5/7) libkcapi-tests.py:LibKCAPI.test;kcapi-enc-test-large.sh-adb4: FAIL: 1 test(s) failed, please refer to the log (13.66 s)
 (6/7) libkcapi-tests.py:LibKCAPI.test;hasher-test.sh-4245: FAIL: 168 test(s) failed, please refer to the log (17.21 s)
 (7/7) libkcapi-tests.py:LibKCAPI.test;kcapi-fuzz-test.sh-613d: FAIL: 3 test(s) failed, please refer to the log (23.85 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 7 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/nasastry/avocado-fvt-wrapper/results/job-2020-10-02T13.32-edd1ec8/results.html
JOB TIME   : 153.96 s
```


SLES:
```
Fetching asset from libkcapi-tests.py:LibKCAPI.test
JOB ID     : ca7171abc13c04e225de41b54437fe3872c0379b
JOB LOG    : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.45-ca7171a/job.log
 (1/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-test.sh-9d21: PASS (211.54 s)
 (2/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-convenience.sh-3236: PASS (18.49 s)
 (3/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-dgst-test.sh-6f25: PASS (30.45 s)
 (4/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test.sh-cb9f: PASS (29.99 s)
 (5/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-enc-test-large.sh-adb4: PASS (18.45 s)
 (6/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-hasher-test.sh-4245: FAIL: 8 test(s) failed, please refer to the log (23.02 s)
 (7/7) libkcapi-tests.py:LibKCAPI.test;run-libkcapi-test-kcapi-fuzz-test.sh-613d: PASS (1029.39 s)
RESULTS    : PASS 6 | ERROR 0 | FAIL 1 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/OpTest/avocado-fvt-wrapper/results/job-2020-10-06T09.45-ca7171a/results.html
JOB TIME   : 1363.06 s
```

**Review by PraveenPenguin — APPROVED** [review]
LGTM

**Pull request merged**